src/ports/postgres/modules/graph/bfs.py_in

```diff
@@ -0,0 +1,445 @@
+                    INSERT INTO {out_table}
+                    SELECT {grp_comma} {vertex_id}, {dist_col}, min({parent_col})
+                    FROM (
+                        SELECT {grp_comma} 
+                            {dest} AS {vertex_id}, 
+                            {curr_dist_val}+1 AS {dist_col}, 
+                            {src} AS {parent_col}
+                        FROM {edge_table} 
+                        WHERE (
+                            ({grp_comma} {src}) IN ( 
+                                SELECT {grp_comma} {vertex_id} FROM {out_table} 
+                                WHERE {dist_col}={curr_dist_val}
+                            ) 
+                            AND 
+                            ({grp_comma} {dest}) NOT IN (
+                                SELECT {grp_comma} {vertex_id} FROM {out_table}
+                            )
+                        )
+                        UNION
+                        SELECT {grp_comma} 
+                            {src} AS {vertex_id}, 
+                            {curr_dist_val}+1 AS {dist_col}, 
+                            {dest} AS {parent_col}
+                        FROM {edge_table} 
+                        WHERE (
+                            ({grp_comma} {dest}) IN (
+                                SELECT {grp_comma} {vertex_id} FROM {out_table} 
+                                WHERE {dist_col}={curr_dist_val}
+                            ) 
+                            AND 
+                            ({grp_comma} {src}) NOT IN (
+                                SELECT {grp_comma} {vertex_id} FROM {out_table}
+                            )
+                        )
+                    ) t1
+                    GROUP BY {grp_comma} {vertex_id}, {dist_col}
+                """.format(**locals()))
+
+                curr_dist_val = curr_dist_val + 1
+                
+                vct = plpy.execute("""
+                    SELECT COUNT(*) 
+                    FROM {edge_table} 
+                    WHERE (
+                        ({grp_comma} {src}) IN ( 
+                            SELECT {grp_comma} {vertex_id} FROM {out_table} 
+                            WHERE {dist_col}={curr_dist_val}
+                        ) 
+                        AND 
+                        ({grp_comma} {dest}) NOT IN (
+                            SELECT {grp_comma} {vertex_id} FROM {out_table}
+                        )
+                    ) OR (
+                        ({grp_comma} {dest}) IN ( 
+                            SELECT {grp_comma} {vertex_id} FROM {out_table} 
+                            WHERE {dist_col}={curr_dist_val}
+                        ) 
+                        AND 
+                        ({grp_comma} {src}) NOT IN (
+                            SELECT {grp_comma} {vertex_id} FROM {out_table}
+                        )
+                    )
+                    """.format(**locals()))[0]['count']
+
```

Comment:
We could probably replace the if-else with a single block here too, similar to the comment on line 312.